.semaphore/semaphore.yml

```diff
@@ -0,0 +1,725 @@
+                
+            - export PATH=$PATH:/usr/local/bin/protoc
+            - export PATH=$PATH:$HOME/go/bin
+             
+            - |
+              echo "Setting up Mainflux..."
+              for p in $(ls users/policies/*.pb.go); do
+                mv $p $p.tmp
+              done
+              for p in $(ls things/policies/*.pb.go); do
+                mv $p $p.tmp
+              done                
+              for p in $(ls pkg/messaging/*.pb.go); do
+                mv $p $p.tmp
+              done
+            - protoc -I. --go_out=. --go_opt=paths=source_relative pkg/messaging/*.proto
+            - protoc -I. --go_out=. --go_opt=paths=source_relative --go-grpc_out=. --go-grpc_opt=paths=source_relative users/policies/*.proto
+            - protoc -I. --go_out=. --go_opt=paths=source_relative --go-grpc_out=. --go-grpc_opt=paths=source_relative things/policies/*.proto
+            - |
+              for p in $(ls users/policies/*.pb.go); do
+                if ! cmp -s $p $p.tmp; then
+                  echo "Proto file and generated Go file $p are out of sync!"
+                  exit 1
+                fi
+              done
+            - |
+              for p in $(ls things/policies/*.pb.go); do
+                if ! cmp -s $p $p.tmp; then
+                  echo "Proto file and generated Go file $p are out of sync!"
+                  exit 1
+                fi
+              done
+            - |                
+              for p in $(ls pkg/messaging/*.pb.go); do
+                if ! cmp -s $p $p.tmp; then
+                  echo "Proto file and generated Go file $p are out of sync!"
+                  exit 1
+                fi
+              done
+  - name: Lint and Test Users
+    run:
+      when: "change_in(['users', 'cmd/users', 'users/policies/auth.proto'])"
+    dependencies:
+      - Setup
+      - Check Generated Protocol Buffer Files
+    task:
+      jobs:
+        - name: Lint Users
+          commands:
+            - cd users
+            - cache restore linter
+            - './bin/golangci-lint run --no-config --disable-all --enable gosimple,errcheck,govet,unused,goconst,godot --timeout 3m'
+        - name: Test Users
+          commands:
+            - cd users
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+      secrets:
+        - name: CODECOV_TOKEN
+  - name: Build Users
+    dependencies:
+      - Lint and Test Users
+    run:
+      when: "change_in(['users', 'cmd/users'])"
+    task:
+      jobs:
+        - name: Build Users
+          commands:
+            - make docker_users
+      secrets:
+        - name: DOCKER_TOKEN            
+        - name: DOCKER_USERNAME                    
+  - name: Lint and Test Things
+    run:
+      when: "change_in(['things', 'cmd/things', 'users/policies/auth.proto'])"
```

Comment:
Let's test if changes are in `/users/policies/postgres/policies.go` and `/users/policies/api/grpc/client.go`, not `user/policies/auth.proto`.